GoogleChromeLabs/ps-analysis-tool
# Feature: Show Exempted cookies in extension

## Description
This PR aims to show the cookie exemption reason from the Chrome DevTools Protocol. These are the cookies which were to be blocked but were exempted due to some reasons.

## Relevant Technical Choices
- Use the CDP `extraInfo` events to get the cookie exemption reasons:
  - [Network.requestWillBeSentExtraInfo.associatedCookies](https://chromedevtools.github.io/devtools-protocol/tot/Network/#event-requestWillBeSentExtraInfo)
- [Network.responseReceivedExtraInfo.exemptedCookies](https://chromedevtools.github.io/devtools-protocol/tot/Network/#event-responseReceivedExtraInfo) 

## Testing Instructions
- Clone this branch.
- Go to the terminal and run `npm start`.
- Open the browser with 3pcd flags enabled.
- Load the extension and open `edition.cnn.com`.
- Go to PSAT in DevTools and open the cookie table.
- Allow list a 3rd party cookie and go to the cookies landing page.
- In the cookies landing page you will see and exempted cookies section with the count and the reasoning.

## Additional Information:
- [Network.CookieExemptionReason](https://chromedevtools.github.io/devtools-protocol/tot/Network/#type-CookieExemptionReason)


## Screenshot/Screencast
LandingPage:
![Screenshot 2024-02-23 at 15 19 06](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/7c59abb0-a21f-4283-b8a7-c671711e9911)


---

## Checklist






Fixes #503
